src/node_file-inl.h

```diff
@@ -349,6 +349,38 @@ int SyncCall(Environment* env, v8::Local<v8::Value> ctx,
   return err;
 }
 
+// Similar to SyncCall but throws immediately if there is an error.
+template <typename Predicate, typename Func, typename... Args>
+int SyncCallAndThrowIf(Predicate should_throw,
+                       Environment* env,
+                       FSReqWrapSync* req_wrap,
+                       Func fn,
+                       Args... args) {
+  env->PrintSyncTrace();
```

Comment:
If the lack of this call causes a bug, I recommend adding a call to address your concerns in the description

Reply:
We do have a test (`test/parallel/test-sync-io-option.js`), but it only checks `fs.statSync`.

Reply:
I unknowingly broke it. We should eventually add it.